### PR TITLE
Adding Plex (OpenPHT) as a port

### PR DESCRIPTION
**Pull request by j-r0dd**
Adding Plex (OpenPHT) as a port. This may change to Plex Media Player if/when Plex ever releases an official standalone app for Linux. Limiting this to x86 for now since that is the only pre-built packages available. The pi versions are embedded system images. Could probably have build from source, but being lazy.

### CODE REVIEW

**Comment by joolswills** [comment]
Why was the second PR opened ?

File: scriptmodules/ports/plex.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+rp_module_section="opt"
+rp_module_flags="!arm"
+
+VERSION="1.6.2.123-e23a7eef"
+APTVER=$(apt --version | cut -d' ' -f2) 
+DISTRO=$(lsb_release -is)
+CODENAME=$(lsb_release -cs)
+RELEASE=$(lsb_release -rs)
+PACKAGE="openpht_${VERSION}-${CODENAME}_amd64.deb"
+GETDEB="https://github.com/RasPlex/OpenPHT/releases/download/v${VERSION}/${PACKAGE}"
```
Comment (joolswills) [inline scriptmodules/ports/plex.sh]: Global vars are not allowed in modules - you can move them to the functions - use lowercase and local vars. You can use __raspbian_name / __raspbian_ver for comparing jessie / version.

File: scriptmodules/ports/plex.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+function install_bin_plex() {
+    if [[ "${DISTRO}" == "Debian" && "${RELEASE}" -lt "8" ]]; then
+        md_ret_errors+=("The Debian package available is only for Jessie")
+        return 1
+    else
+        wget -nv -O "$__tmpdir/${PACKAGE}" ${GETDEB}
+        if [[ $(dpkg --compare-versions "${APTVER}" ge "1.1"; echo $?) -eq 0  ]]; then
```
Comment (joolswills) [inline scriptmodules/ports/plex.sh]: `hasPackage NAME VER OPERATOR`

File: scriptmodules/ports/plex.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+function remove_plex() {
+    aptRemove openpht
+    rp_callModule plex depends remove
+    apt-get autoremove --purge -y
+}
+
+function configure_plex() {
+    # remove old directLaunch entry
+    delSystem "$md_id" "plex"    
+    rmDirExists "$romdir/plex"
```
Comment (joolswills) [inline scriptmodules/ports/plex.sh]: no need for these two lines.

File: scriptmodules/ports/plex.sh [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+#    fi
+#done < <(tail --pid=$$ -f -n0 $LOG_FILE)
+#
+#killall openpht
+#EOF
+#
+#    chmod +x "$romdir/plex/Plex.sh"
+#
+#    setESSystem 'Plex' 'plex' '~/RetroPie/roms/plex' '.sh .SH' '%ROM%' 'plex' 'plex'
+#}
```
Comment (joolswills) [inline scriptmodules/ports/plex.sh]: This should be removed.

**Comment by joolswills** [comment]
Thanks. I have added some comments as there will need to be a few changes before this can be merged.

File: scriptmodules/ports/plex.sh [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            # I feel dirty
+            dpkg -i $__tmpdir/${PACKAGE}
+            apt-get -f -y install
+        fi
+    fi
+}
+
+function remove_plex() {
+    aptRemove openpht
+    rp_callModule plex depends remove
```
Comment (joolswills) [inline scriptmodules/ports/plex.sh]: the module has no depends function so this is not needed.

**Comment by joolswills** [comment]
I think if the project is called "openpht", the module should be perhaps be called that too, to distinguish itself from plex media player etc.

**Comment by j-r0dd** [comment]
Alright I will clean this up a bit and change to openpht instead of plex. Will open a new pull request when I get time to fix.

**Comment by joolswills** [comment]
Please don't do that - just modify the code and push it to the branch - you can squash/force push as needed or I can do a final squash at the end.

**Comment by j-r0dd** [comment]
it looks like a may have squashed the wrong commits. i really suck at git.

**Comment by j-r0dd** [comment]
ok i think i git it all straightened out this.

File: scriptmodules/ports/openpht.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    if [[ "$__raspbian_name" == "wheezy"  ]]; then
+        md_ret_errors+=("The Debian package available is only for Jessie")
+        return 1
+    else
+        wget -nv -O "$__tmpdir/$package" $getdeb
+        if hasPackage "apt" "1.1" "ge"; then
+            apt install -y --allow-downgrades $__tmpdir/$package
+        else
+            # I feel dirty
```
Comment (joolswills) [inline scriptmodules/ports/openpht.sh]: Although amusing, a better comment would be "falling back to dpkg" or similar no ? :)

File: scriptmodules/ports/openpht.sh [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+rp_module_id="openpht"
+rp_module_desc="OpenPHT is a community driven fork of Plex Home Theater"
+rp_module_section="opt"
+rp_module_flags="!arm"
+
+function install_bin_openpht() {
+local version="1.6.2.123-e23a7eef"
+local codename=$(lsb_release -cs)
+local package="openpht_${version}-${codename}_amd64.deb"
+local getdeb="https://github.com/RasPlex/OpenPHT/releases/download/v$version/$package"
```
Comment (joolswills) [inline scriptmodules/ports/openpht.sh]: wrong indent

**Comment by j-r0dd** [comment]
Alright hopefully that will take care of it

**Comment by joolswills** [comment]
Thanks.

**Comment by joolswills** [comment]
aah oops. it wasn't completely squashed and I didn''t squash on merge. ah well.

**Pull request merged**